WasatchNET/EEPROMJSON.cs

```diff
@@ -0,0 +1,425 @@
+            {
+                if (item.ROIVertRegionStarts[i] != this.ROIVertRegionStarts[i])
+                    return false;
+            }
+            //public double[] ROIVertRegionEnds;
+            if (item.ROIVertRegionEnds.Length != this.ROIVertRegionEnds.Length)
+                return false;
+            for (int i = 0; i < this.ROIVertRegionEnds.Length; ++i)
+            {
+                if (item.ROIVertRegionEnds[i] != this.ROIVertRegionEnds[i])
+                    return false;
+            }
+            if (item.LaserPowerCoeffs.Length != this.LaserPowerCoeffs.Length)
+                return false;
+            for (int i = 0; i < this.LaserPowerCoeffs.Length; ++i)
+            {
+                if (Math.Abs(item.LaserPowerCoeffs[i] - this.LaserPowerCoeffs[i]) > 0.00001)
+                    return false;
+            }
+            //public double MaxLaserPowerMW;
+            if (Math.Abs(item.MaxLaserPowerMW - this.MaxLaserPowerMW) > 0.00001)
+                return false;
+            //public double MinLaserPowerMW;
+            if (Math.Abs(item.MinLaserPowerMW - this.MinLaserPowerMW) > 0.00001)
+                return false;
+            //public double ExcitationWavelengthNM;
+            if (Math.Abs(item.ExcitationWavelengthNM - this.ExcitationWavelengthNM) > 0.00001)
+                return false;
+            if (Math.Abs(item.AvgResolution - this.AvgResolution) > 0.00001)
+                return false;
+            //public double[] BadPixels;
+            if (item.BadPixels.Length != this.BadPixels.Length)
+                return false;
+            for (int i = 0; i < this.BadPixels.Length; ++i)
+            {
+                if (item.BadPixels[i] != this.BadPixels[i])
+                    return false;
+            }
+            //public string UserText;
+            if (item.UserText != this.UserText)
+                return false;
+            //public string ProductConfig;
+            if (item.ProductConfig != this.ProductConfig)
+                return false;
+            //public int RelIntCorrOrder;
+            if (item.RelIntCorrOrder != this.RelIntCorrOrder)
+                return false;
+            //public double[] RelIntCorrCoeff;
+            if (item.RelIntCorrCoeffs != null)
+            {
+                if (item.RelIntCorrCoeffs.Length != this.RelIntCorrCoeffs.Length)
+                    return false;
+                for (int i = 0; i < this.RelIntCorrCoeffs.Length; ++i)
+                {
+                    if (Math.Abs(item.RelIntCorrCoeffs[i] - this.RelIntCorrCoeffs[i]) > 0.00001)
+                        return false;
+                }
+            }
+            //public bool Bin2x2;
+            //public bool FlipXAxis;
+            if (item.Bin2x2 != this.Bin2x2)
+                return false;
+            if (item.FlipXAxis != this.FlipXAxis)
+                return false;
+
+            return true;
+        }
+
+        public override int GetHashCode()
```

Comment:
I've never made one of these, so can't comment on it...but it seems weird.

Reply:
Now THAT one is auto generated